pennylane/templates/embeddings/amplitude.py

```diff
@@ -211,13 +211,7 @@ def _preprocess(features, wires, pad_with, normalize):
                 # pad
                 if n_features < dim:
                     padding = [pad_with] * (dim - n_features)
-                    if (
-                        hasattr(feature_set, "device") and feature_set.device.type == "cuda"
-                    ):  # pragma: no cover
-                        ## Torch tensor, send to same GPU
-                        dat_type = type(feature_set)
-                        padding = dat_type(padding).to(feature_set.device)
-
+                    padding = qml.math.convert_like(padding, feature_set)
```

Comment:
Does it also work with GPU with Torch? I added the GPU tests

Reply:
Yes, `qml.math.convert_like` already moves torch tensors to the correct device.